### DISCUSSION TITLE
Reason for 30ms Task.Delay in DataReceiver?

### DISCUSSION THREAD

**Discussion by derMaaster** [jchristn/WatsonTcp#294] [General]
Hi,

Within WatsonTcpClient.cs line 770 there is this code in the read-message region:

`                    await _ReadLock.WaitAsync(token);
                    WatsonMessage msg = await _MessageBuilder.BuildFromStream(_DataStream, token);
                    if (msg == null)
                    { 
                        await Task.Delay(30, token).ConfigureAwait(false);
                        continue;
                    }`

Can someone please explain the logic of the above code specific to the Task.Delay of 30 milliseconds?

With that asked, it is mentioned in the README that WatsonTcp is the fastest, meaning it is a latency sensitive / optimized library?

**Comment by jchristn** [comment]
This `Task.Delay` is used to create a pause before checking (again) for another available message.  Since framing is involved, it wouldn't make sense to immediately `continue` (framing data has to be deserialized, data has to be read from the socket, and so on).  

When I claim it's the "fastest", it means the amount of time it takes a developer to get a TCP server and client up and running :)

**Comment by derMaaster** [comment]
Many thanks Joel.
